internals/errio/errors.go

```diff
@@ -105,17 +105,10 @@ func StatusError(err error) error {
 	return UnexpectedStatusError(err)
 }
 
-// Error can be called to on any error to convert it to a PublicError if it is not already.
-// If it is not yet a PublicError, an UnexpectedError is returned
+// Error returns the inputted error.
+// Deprecated: this leads to unwanted returns of UnexpectedErrrors and should therefor not be used anymore.
 func Error(err error) error {
-	if err == nil {
-		return nil
-	}
-
-	if isPublicError(err) || isPublicStatusError(err) {
-		return err
-	}
-	return UnexpectedError(err)
+	return err
```

Comment:
Not sure if we should change the behavior as well. Should we just deprecate and stop calling the function where we need the changed behavior?

One might depend on the returned type being `UnexpectedError`, as this was explicitly documented and a separate test case exists for this, `TestUnexpectedError`.